mlir/lib/Quantum/IR/QuantumOps.cpp

```diff
@@ -58,7 +58,8 @@ LogicalResult CustomOp::canonicalize(CustomOp op, mlir::PatternRewriter &rewrite
 
             rewriter.replaceOpWithNewOp<CustomOp>(
                 op, op.getOutQubits().getTypes(), op.getOutCtrlQubits().getTypes(), paramsNeg,
-                op.getInQubits(), name, nullptr, op.getInCtrlQubits(), op.getInCtrlValues());
+                op.getInQubits(), name, nullptr, op.getInCtrlQubits(), op.getInCtrlValues(),
+                nullptr);
```

Comment:
The purpose of this section is to negate the angles during rotation op's canonicalization. We should update the static path as well instead of just putting nullptr here.

There might be two cases here:
1. Param comes in dynamically, and the optional static param attr does not exist. This case works fine as is.
2. Param somes in statically, and the variadic dyn_params has zero length. In this case this rewrite will completely void the params, as `paramsNeg` will have zero length, and static param is nullptr.

Reply:
Yes, I know this is an issue.I am working on this part.